### DISCUSSION TITLE
Default Value Properties

### DISCUSSION THREAD

**Discussion by TRexStark** [leekelleher/umbraco-contentment#309] [Idea]
Hi @leekelleher 

A while ago I created a Custom Property for a string and a textarea that you can add default values to. Helps the content editor to have preset values when necessary.

Given Contentment is a rich featured Package, it'll only make sense to have this feature part of it.

Initially, I also had a toggle to make them read only, but after I discovered the Admin Only Property https://github.com/LottePitcher/umbraco-admin-only-property I didn't need the readonly feature so I wasn't using it in my projects anymore.

Would you be interested on a Pull Request or I can just send you the code?

The only thing I couldn't figure out is this:
If you already have a content node with these properties and have existing values, then if you edit the Document Type, the original default values overwrites the value in the content node if it has been changed by the editor - maybe I can leave that for you to tweak? Of course if you have interested in implementing this.

**Comment by leekelleher** [comment]
@TRexStark Thank you for the suggestion, but default values is not something I'm looking to add to Contentment's property-editors. _Although I'm frequently asked about this._

The main reason being is that I don't see the property-editor or Data Type as the place to configure a default value, it should be done either at Content Type level or use a Content Template (blueprint).

Personally, for the times where I have needed to add a default value, I've ended up hooking into the `SendingContentNotification` notification handler programmatically.

Ultimately, I believe this is a pattern that Umbraco core needs to solve, not at property-editor / Data Type level.

**Reply by ault** [reply]
Whilst I would possibly agree where it comes to strings and text areas, I find that a default value for data lists (radio buttons, select lists, etc) is essential. A radio button list without a default checked item is worse than bad ux. Selecting a default option at the data type level seems like the natural place for it.

**Reply by ault** [reply]
I've seen the other discussions about where to place the default option and the problems that it raises. From talks with Dean Leigh it seems that Umbraco central blames Angular and doesn't want to touch it.

It's my firm opinion that in the list editor the user should see a tickbox next to each data list item (in addition to icon, value, description, etc) to make it a default item. If the user selects several and uses the list as radio buttons, it's their fault. If used as a select list or group of tickboxes it makes perfect sense.

**Comment by TRexStark** [comment]
Thanks @leekelleher I agree with you.

**Comment by jesperordrup** [comment]
In a situation where data is static it would be useful to be able to add an equally static default value to the configuration. Optionally of course.

**Comment by TRexStark** [comment]
@jesperordrup / @ault, apparently it's not as straight forward as it seems. I previously created a custom string DataType to do this, the problem is, when you set a default value that editors can change, once you update/change the doc type that holds this data type, the value reverts back to the default value, which means any changes the editor made would be overwritten by you.

I haven't put any thoughts on how to prevent that yet. But will keep thinking.

**Comment by leekelleher** [comment]
Closing this one off, same reasons as https://github.com/leekelleher/umbraco-contentment/discussions/91#discussioncomment-17017175, as we have the [`propertyValuePreset` extension point](https://docs.umbraco.com/umbraco-cms/extend-your-project/backoffice-extensions/extending-overview/extension-types/property-value-preset) since v15.
